src/indexing/codecs/residual.jl

```diff
@@ -782,3 +782,13 @@ function decompress(
     end
     embeddings
 end
+
+function Base.show(io::IO, codec::Dict{String, Any})
```

Comment:
I think you don't need a method to print the `codec`; it's just a bunch of matrices anyways, and I doubt anyone would want to `show` it explicitly.